allauth/socialaccount/providers/battlenet/provider.py

```diff
@@ -5,17 +5,13 @@
 
 
 class BattleNetSocialAccountAdapter(DefaultSocialAccountAdapter):
```

Comment:
I must admit, I completely missed out on this class when reviewing and merging the provider. It is uncommon to provide a custom adapter with a provider. Adapters are meant to be generic, mostly provider independent. If you need to influence the way the username is chosen, then make sure that `extract_common_fields` of the `BattleNetProvider` returns a `'username'` key. So let's drop this class completely, and move the username logic over there...